apiserver/apiserver.go

```diff
@@ -1060,12 +1060,13 @@ func (srv *Server) processModelRemovals() error {
 			return tomb.ErrDying
 		case modelUUIDs := <-w.Changes():
 			for _, modelUUID := range modelUUIDs {
-				model, err := st.GetModel(names.NewModelTag(modelUUID))
+				model, release, err := srv.statePool.GetModel(modelUUID)
 				gone := errors.IsNotFound(err)
 				dead := err == nil && model.Life() == state.Dead
 				if err != nil && !gone {
 					return errors.Trace(err)
 				}
+				release()
```

Comment:
At this point, err could be not found. Will release() be valid then? Don't we want to check that err == nil?